dvc/analytics.py

```diff
@@ -1,256 +1,120 @@
-        """
-        import tempfile
 
-        with tempfile.NamedTemporaryFile(delete=False, mode="w") as fobj:
-            json.dump(self.info, fobj)
-            return fobj.name
+def system_info():
+    system = platform.system()
 
-    @staticmethod
-    def is_enabled(cmd=None):
-        from dvc.config import Config, to_bool
-        from dvc.command.daemon import CmdDaemonBase
+    if system == "Windows":
+        version = sys.getwindowsversion()
 
-        if env2bool("DVC_TEST"):
-            return False
+        return {
+            "os": "windows",
+            "windows_version_build": version.build,
+            "windows_version_major": version.major,
+            "windows_version_minor": version.minor,
+            "windows_version_service_pack": version.service_pack,
+        }
 
-        if isinstance(cmd, CmdDaemonBase):
-            return False
+    if system == "Darwin":
+        return {"os": "mac", "mac_version": platform.mac_ver()[0]}
 
-        core = Config(validate=False).config.get(Config.SECTION_CORE, {})
-        enabled = to_bool(core.get(Config.SECTION_CORE_ANALYTICS, "true"))
-        logger.debug(
-            "Analytics is {}.".format("enabled" if enabled else "disabled")
-        )
-        return enabled
+    if system == "Linux":
+        return {
+            "os": "linux",
+            "linux_distro": distro.id(),
+            "linux_distro_like": distro.like(),
+            "linux_distro_version": distro.version(),
+        }
 
-    @staticmethod
-    def send_cmd(cmd, args, ret):
-        """Collect and send analytics for CLI command.
+    return {"os": system.lower()}
 
-        Args:
-            args (list): parsed args for the CLI command.
-            ret (int): return value of the CLI command.
-        """
-        from dvc.daemon import daemon
 
-        if not Analytics.is_enabled(cmd):
-            return
+def find_or_create_user_id():
+    """
+    The user's ID is stored on a file under the global config directory.
 
-        analytics = Analytics()
-        analytics.collect_cmd(args, ret)
-        daemon(["analytics", analytics.dump()])
+    The file should contain a JSON with a "user_id" key:
 
-    def send(self):
-        """Collect and send analytics."""
-        import requests
+        {"user_id": "16fd2706-8baf-433b-82eb-8c7fada847da"}
 
-        if not self.is_enabled():
-            return
+    IDs are generated randomly with UUID.
+    """
+    config_dir = Config.get_global_config_dir()
+    fname = config_dir / "user_id"
+    lockfile = fname.with_suffix(".lock")
 
-        self.collect()
+    try:
+        with Lock(lockfile):
+            try:
+                user_id = json.loads(fname.read_text())["user_id"]
+            except (FileNotFoundError, ValueError, AttributeError):
+                user_id = str(uuid.uuid4())
+                makedirs(fname.parent, exist_ok=True)
+                fname.write_text(str(json.dumps({"user_id": user_id})))
```

Comment:
Why do we need this `str`? Doesn't it already return str? Or is it py2 compatibility?

Reply:
py2 compatibility, `unicode` stuff :/

Reply:
@mroutis so write_text requires unicode, right? In that case, use our compat function to do that instead of str. I think `convert_to_unicode` is suitable here. We need to pay attention to such things, so that we can easilly get rid of them when py2 reaches EOL.

Reply:
@efiop , agree!